### PR TITLE
Add assign_permission_sets task

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

- CumulusCI offers an `assign_permission_sets` task, which assigns one or more Permission Sets to the running user if not already assigned. This task may replace invocations of `dx`.

# Issues Closed

### CODE REVIEW

**Review by Br4nd0R — DISMISSED** [review]
LGTM!

File: cumulusci/tasks/salesforce/users/permsets.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        self.options["api_names"] = process_list_arg(self.options["api_names"])
+
+    def _run_task(self):
+        # Determine existing assignments
+        query = f"""SELECT Id,
+                           (SELECT PermissionSetId
+                            FROM PermissionSetAssignments)
+                    FROM User
+                    WHERE Id = '{self.org_config.user_id}'"""
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/users/permsets.py:25]: For scratch orgs, `user_id` is a property which does its own query of User filtering by username. Would it make more sense to just filter by username in this query?

File: cumulusci/tasks/salesforce/users/permsets.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        # Assign all not already assigned
+        for api_name in self.options["api_names"]:
+            if permsets[api_name] not in assigned_permsets:
+                self.sf.PermissionSetAssignment.create(
+                    {
+                        "AssigneeId": self.org_config.user_id,
+                        "PermissionSetId": permsets[api_name],
+                    }
+                )
```
Comment (davisagli) [inline cumulusci/tasks/salesforce/users/permsets.py:53]: Log which permsets are actually getting assigned

**Review by davisagli — CHANGES_REQUESTED** [review]

**Review by davisagli — DISMISSED** [review]

**Comment by davidmreed** [comment]
After discussion with @bethbrains added ability to target a user by alias.

**Comment by bethbrains** [comment]
Hooray! Thank you!

**Review by davisagli — APPROVED** [review]

**Pull request merged**